test/pkg/integration/workspace.go

```diff
@@ -123,6 +126,11 @@ func LaunchWorkspaceDirectly(t *testing.T, ctx context.Context, api *ComponentAP
 		return nil, nil, err
 	}
 
+	parallelLimiter <- struct{}{}
```

Comment:
Can you not use go test's `-parallel` flag?

Reply:
Actually, that is what I tried first. However, in that case, the cleanup process runs asynchronously and goes to the next test. In other words, the workspace stop process goes to the next test before it completes, which increases the load too much. Therefore, we did not use it.